src/main/java/com/netflix/imflibrary/reader_interfaces/MXFEssenceReader.java

```diff
@@ -180,6 +180,23 @@ else if(headerPartition.hasWaveAudioEssenceDescriptor()){
     }
 
     /**
+     * A method that returns the EssenceType enumeration corresponding to this Essence
+     * @return a string representing either a MainImageSequence or a MainAudioSequence
+     * @throws IOException - any I/O related error will be exposed through an IOException
+     */
+    public CompositionPlaylist.SequenceTypeEnum getEssenceTypeEnum() throws IOException{
+        CompositionPlaylist.SequenceTypeEnum result = CompositionPlaylist.SequenceTypeEnum.Unknown;
+        HeaderPartition headerPartition = this.getHeaderPartitionIMF().getHeaderPartitionOP1A().getHeaderPartition();
+        if(headerPartition.hasCDCIPictureEssenceDescriptor() || headerPartition.hasRGBAPictureEssenceDescriptor()){
```

Comment:
Isn't better to add in HeaderPartition class a Enum type and getter to describe essence category like Video/Audio/Subtitle/etc.
which allow you to replace this code with a switch/case state and throw an exception in default case with something like "unsupported essence type".